### PR TITLE
feat(plasma-b2c): processing "Text Field" view

### PR DESCRIPTION
**Pull request by Yakutoc**
:baby_chick: Download canary assets:

[colors--canary.244.3746513770.xml](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/colors--canary.244.3746513770.xml)  
[color_metro_ios-swift--canary.244.3746513770.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_metro_ios-swift--canary.244.3746513770.swift)  
[color_metro_kotlin--canary.244.3746513770.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_metro_kotlin--canary.244.3746513770.kt)  
[color_metro_react-native--canary.244.3746513770.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_metro_react-native--canary.244.3746513770.ts)  
[color_sbermarket_ios-swift--canary.244.3746513770.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sbermarket_ios-swift--canary.244.3746513770.swift)  
[color_sbermarket_kotlin--canary.244.3746513770.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sbermarket_kotlin--canary.244.3746513770.kt)  
[color_sbermarket_react-native--canary.244.3746513770.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sbermarket_react-native--canary.244.3746513770.ts)  
[color_sberprime_ios-swift--canary.244.3746513770.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sberprime_ios-swift--canary.244.3746513770.swift)  
[color_sberprime_kotlin--canary.244.3746513770.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sberprime_kotlin--canary.244.3746513770.kt)  
[color_sberprime_react-native--canary.244.3746513770.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sberprime_react-native--canary.244.3746513770.ts)  
[color_selgros_ios-swift--canary.244.3746513770.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_selgros_ios-swift--canary.244.3746513770.swift)  
[color_selgros_kotlin--canary.244.3746513770.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_selgros_kotlin--canary.244.3746513770.kt)  
[color_selgros_react-native--canary.244.3746513770.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_selgros_react-native--canary.244.3746513770.ts)  
[color_smbusiness_ios-swift--canary.244.3746513770.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_smbusiness_ios-swift--canary.244.3746513770.swift)  
[color_smbusiness_kotlin--canary.244.3746513770.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_smbusiness_kotlin--canary.244.3746513770.kt)  
[color_smbusiness_react-native--canary.244.3746513770.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_smbusiness_react-native--canary.244.3746513770.ts)  
[PlasmaTokensColor--canary.244.3746513770.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/PlasmaTokensColor--canary.244.3746513770.swift)  
[typo_mage_kotlin--canary.244.3746513770.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_mage_kotlin--canary.244.3746513770.kt)  
[typo_mage_react-native--canary.244.3746513770.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_mage_react-native--canary.244.3746513770.ts)  
[typo_plasma_kotlin--canary.244.3746513770.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_plasma_kotlin--canary.244.3746513770.kt)  
[typo_plasma_react-native--canary.244.3746513770.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_plasma_react-native--canary.244.3746513770.ts)  
[typo_ruler_kotlin--canary.244.3746513770.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_ruler_kotlin--canary.244.3746513770.kt)  
[typo_ruler_react-native--canary.244.3746513770.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_ruler_react-native--canary.244.3746513770.ts)  
[typo_sage_kotlin--canary.244.3746513770.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_sage_kotlin--canary.244.3746513770.kt)  
[typo_sage_react-native--canary.244.3746513770.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_sage_react-native--canary.244.3746513770.ts)  
[typo_sbermarket_kotlin--canary.244.3746513770.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_sbermarket_kotlin--canary.244.3746513770.kt)  
[typo_sbermarket_react-native--canary.244.3746513770.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_sbermarket_react-native--canary.244.3746513770.ts)  
[typo_soulmate_kotlin--canary.244.3746513770.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_soulmate_kotlin--canary.244.3746513770.kt)  
[typo_soulmate_react-native--canary.244.3746513770.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_soulmate_react-native--canary.244.3746513770.ts)

<details>
  <summary>📦 Published PR as canary version: <code>Canary Versions</code></summary>
  <br />

  :sparkles: Test out this PR locally via:
  
  ```bash
  npm install @salutejs/plasma-b2c@1.121.0-canary.244.3746513770.0
  # or 
  yarn add @salutejs/plasma-b2c@1.121.0-canary.244.3746513770.0
  ```
</details>

### CODE REVIEW

**Comment by Salute-Eva** [comment]
Demo canvas app deployed!
  
  http://plasma.sberdevices.ru/demo-canvas-app-pr-244/
  <details>
    <summary>New documentation</summary>
    <br />

    http://plasma.fine.sberdevices.ru/pr/demo-canvas-app-pr-244/

  </details>

**Comment by Salute-Eva** [comment]
Documentation preview deployed!
  
  website: http://plasma.sberdevices.ru/pr-244/
  ui storybook: http://plasma.sberdevices.ru/pr-244/ui-storybook/
  web storybook: http://plasma.sberdevices.ru/pr-244/web-storybook/
  b2c storybook: http://plasma.sberdevices.ru/pr-244/b2c-storybook/
  temple storybook: http://plasma.sberdevices.ru/pr-244/temple-storybook/
  <details>
    <summary>New documentation</summary>
    <br />

    website: http://plasma.fine.sberdevices.ru/pr/pr-244/
    ui storybook: http://plasma.fine.sberdevices.ru/pr/pr-244/ui-storybook/
    web storybook: http://plasma.fine.sberdevices.ru/pr/pr-244/web-storybook/
    b2c storybook: http://plasma.fine.sberdevices.ru/pr/pr-244/b2c-storybook/
    temple storybook: http://plasma.fine.sberdevices.ru/pr/pr-244/temple-storybook/

  </details>

**Comment by Salute-Eva** [comment]
Demo canvas app deployed!
  
  http://plasma.sberdevices.ru/demo-canvas-app-pr-244/
  <details>
    <summary>New documentation</summary>
    <br />

    http://plasma.fine.sberdevices.ru/pr/demo-canvas-app-pr-244/

  </details>

**Comment by Salute-Eva** [comment]
Documentation preview deployed!
  
  website: http://plasma.sberdevices.ru/pr-244/
  ui storybook: http://plasma.sberdevices.ru/pr-244/ui-storybook/
  web storybook: http://plasma.sberdevices.ru/pr-244/web-storybook/
  b2c storybook: http://plasma.sberdevices.ru/pr-244/b2c-storybook/
  temple storybook: http://plasma.sberdevices.ru/pr-244/temple-storybook/
  <details>
    <summary>New documentation</summary>
    <br />

    website: http://plasma.fine.sberdevices.ru/pr/pr-244/
    ui storybook: http://plasma.fine.sberdevices.ru/pr/pr-244/ui-storybook/
    web storybook: http://plasma.fine.sberdevices.ru/pr/pr-244/web-storybook/
    b2c storybook: http://plasma.fine.sberdevices.ru/pr/pr-244/b2c-storybook/
    temple storybook: http://plasma.fine.sberdevices.ru/pr/pr-244/temple-storybook/

  </details>

**Comment by Salute-Eva** [comment]
Demo canvas app deployed!
  
  http://plasma.sberdevices.ru/demo-canvas-app-pr-244/
  <details>
    <summary>New documentation</summary>
    <br />

    http://plasma.fine.sberdevices.ru/pr/demo-canvas-app-pr-244/

  </details>

**Comment by Salute-Eva** [comment]
Documentation preview deployed!
  
  website: http://plasma.sberdevices.ru/pr-244/
  ui storybook: http://plasma.sberdevices.ru/pr-244/ui-storybook/
  web storybook: http://plasma.sberdevices.ru/pr-244/web-storybook/
  b2c storybook: http://plasma.sberdevices.ru/pr-244/b2c-storybook/
  temple storybook: http://plasma.sberdevices.ru/pr-244/temple-storybook/
  <details>
    <summary>New documentation</summary>
    <br />

    website: http://plasma.fine.sberdevices.ru/pr/pr-244/
    ui storybook: http://plasma.fine.sberdevices.ru/pr/pr-244/ui-storybook/
    web storybook: http://plasma.fine.sberdevices.ru/pr/pr-244/web-storybook/
    b2c storybook: http://plasma.fine.sberdevices.ru/pr/pr-244/b2c-storybook/
    temple storybook: http://plasma.fine.sberdevices.ru/pr/pr-244/temple-storybook/

  </details>

**Review by neretin-trike — APPROVED** [review]

**Comment by Salute-Eva** [comment]
Demo canvas app deployed!
  
  http://plasma.sberdevices.ru/pr/demo-canvas-app-pr-244/

**Comment by Salute-Eva** [comment]
Documentation preview deployed!
  
  website: http://plasma.sberdevices.ru/pr/pr-244/
  ui storybook: http://plasma.sberdevices.ru/pr/pr-244/ui-storybook/
  web storybook: http://plasma.sberdevices.ru/pr/pr-244/web-storybook/
  b2c storybook: http://plasma.sberdevices.ru/pr/pr-244/b2c-storybook/
  temple storybook: http://plasma.sberdevices.ru/pr/pr-244/temple-storybook/

**Comment by Salute-Eva** [comment]
Documentation preview deployed!
  
  website: http://plasma.sberdevices.ru/pr/pr-244/
  ui storybook: http://plasma.sberdevices.ru/pr/pr-244/ui-storybook/
  web storybook: http://plasma.sberdevices.ru/pr/pr-244/web-storybook/
  b2c storybook: http://plasma.sberdevices.ru/pr/pr-244/b2c-storybook/
  temple storybook: http://plasma.sberdevices.ru/pr/pr-244/temple-storybook/

File: packages/plasma-core/src/components/TextField/TextField.component-test.tsx [hunk 1]
```diff
@@ -22,19 +22,44 @@ describe('plasma-core: TextField', () => {
...
+                <TextFieldStub size="s" />
+                <SpaceMe />
+                <TextFieldStub size="xs" />
             </CypressTestDecorator>,
         );
 
         cy.matchImageSnapshot();
     });
 
-    it('default: handle size', () => {
```
Comment (Yeti-or) [inline packages/plasma-core/src/components/TextField/TextField.component-test.tsx:35]: ну и вопрос если тесты у нас поменялись в core
почему у нас нет новых скриншотов для ui / web

Comment (Yeti-or) [reply packages/plasma-core/src/components/TextField/TextField.component-test.tsx:35]: и почему старые скриншоты в b2c вот например не удалились ?

**Comment by Salute-Eva** [comment]
Demo canvas app deployed!
  
  http://plasma.sberdevices.ru/pr/demo-canvas-app-pr-244/

**Comment by Salute-Eva** [comment]
Documentation preview deployed!
  
  website: http://plasma.sberdevices.ru/pr/pr-244/
  ui storybook: http://plasma.sberdevices.ru/pr/pr-244/ui-storybook/
  web storybook: http://plasma.sberdevices.ru/pr/pr-244/web-storybook/
  b2c storybook: http://plasma.sberdevices.ru/pr/pr-244/b2c-storybook/
  temple storybook: http://plasma.sberdevices.ru/pr/pr-244/temple-storybook/

File: packages/plasma-core/src/components/TextField/TextField.component-test.tsx [hunk 2]
```diff
@@ -22,19 +22,44 @@ describe('plasma-core: TextField', () => {
         );
     }
 
-    it('default', () => {
```
[the remark this answers is not in the document]
Comment (Yakutoc) [reply packages/plasma-core/src/components/TextField/TextField.component-test.tsx:25]: @Yeti-or 

тут скорее я кратко описываю, что делает тест. 

Нам важно что бы скриншот был с коротким именем?

File: packages/plasma-core/src/components/TextField/TextField.component-test.tsx (hunk 1 above)
Comment (Yakutoc) [reply packages/plasma-core/src/components/TextField/TextField.component-test.tsx:35]: Хороший вопрос, смотрел локально 

`npm run cy:ui:run-ct-update-diffs`  

новые скрины не делает.

Comment (Yakutoc) [reply packages/plasma-core/src/components/TextField/TextField.component-test.tsx:35]: > и почему старые скриншоты в b2c вот например не удалились ?

Спасибо, удалю старые.

File: packages/plasma-core/src/components/TextField/TextField.component-test.tsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (Yeti-or) [reply packages/plasma-core/src/components/TextField/TextField.component-test.tsx:25]: Ну в целом не важно, просто остальные тесты были плюс минус такие

File: packages/plasma-core/src/components/TextField/TextField.component-test.tsx (hunk 1 above)
Comment (Yeti-or) [reply packages/plasma-core/src/components/TextField/TextField.component-test.tsx:35]: В web тоже не делает ?

**Comment by Salute-Eva** [comment]
Demo canvas app deployed!
  
  http://plasma.sberdevices.ru/pr/demo-canvas-app-pr-244/

**Comment by Salute-Eva** [comment]
Documentation preview deployed!
  
  website: http://plasma.sberdevices.ru/pr/pr-244/
  ui storybook: http://plasma.sberdevices.ru/pr/pr-244/ui-storybook/
  web storybook: http://plasma.sberdevices.ru/pr/pr-244/web-storybook/
  b2c storybook: http://plasma.sberdevices.ru/pr/pr-244/b2c-storybook/
  temple storybook: http://plasma.sberdevices.ru/pr/pr-244/temple-storybook/

File: packages/plasma-b2c/src/components/TextField/Placeholder.tsx [hunk 3]
```diff
@@ -1,8 +1,14 @@
...
 
-export const Placeholder = styled(FieldPlaceholder)<{ visible?: boolean }>`
-    --display: ${({ visible }) => (visible ? 'inline-flex' : 'none')};
+export const Placeholder = styled(FieldPlaceholder)<{ visible?: boolean; withCaption: boolean }>`
+    --display: ${({ visible, withCaption }) => {
+        if (withCaption) {
+            return 'none';
+        }
+
+        return visible ? 'inline-flex' : 'none';
```
Comment (neretin-trike) [inline packages/plasma-b2c/src/components/TextField/Placeholder.tsx:10]: *минор*

можно чуть упростить:

```
        if (withCaption || !visible) {
            return 'none';
        }

        return 'inline-flex';
```

**Review by neretin-trike — APPROVED** [review]

**Comment by Salute-Eva** [comment]
Demo canvas app deployed!
  
  http://plasma.sberdevices.ru/pr/demo-canvas-app-pr-244/

**Comment by Salute-Eva** [comment]
Documentation preview deployed!
  
  website: http://plasma.sberdevices.ru/pr/pr-244/
  ui storybook: http://plasma.sberdevices.ru/pr/pr-244/ui-storybook/
  web storybook: http://plasma.sberdevices.ru/pr/pr-244/web-storybook/
  b2c storybook: http://plasma.sberdevices.ru/pr/pr-244/b2c-storybook/
  temple storybook: http://plasma.sberdevices.ru/pr/pr-244/temple-storybook/

**Review by Yeti-or — APPROVED** [review]

**Pull request merged**